### PR TITLE
htmldoc: add darwin support

### PR DESCRIPTION
**Pull request by shanemikel**
I also added a long description and updated the homepage to point to
the author’s new github.io site.

Please be patient as the first attempt at contributing packaging for Nixpkgs.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by shanemikel** [comment]
I'm curious how I was able to remove so many build inputs and still get a working build...

**Comment by Mic92** [comment]
It seems that the project maintainer vendorize all libraries now instead: https://github.com/michaelrsweet/htmldoc/tree/master/jpeg

File: pkgs/tools/typesetting/htmldoc/default.nix [hunk 1]
```diff
@@ -6,12 +6,13 @@
...
-  name = "htmldoc-1.8.29";
+stdenv.mkDerivation rec {
+  version = "1.8.29";
+  name = "htmldoc-${version}";
   src = fetchurl {
     url = "https://github.com/michaelrsweet/htmldoc/releases/download"
-        + "/release-1.8.29/htmldoc-1.8.29-source.tar.gz";
-    md5 = "14d32bd772e2bc6af7b9b2233724c3ec";
+      + "/release-${version}/htmldoc-${version}-source.tar.gz";
+    sha256 = "15x0xdf487j4i4gfap5yr83airxnbp2v4lxaz79a4s3iirrq39p0";
```
Comment (Mic92) [inline pkgs/tools/typesetting/htmldoc/default.nix]: we do not use md5 anymore due known hash collision attacks.

File: pkgs/tools/typesetting/htmldoc/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+stdenv.mkDerivation rec {
+  version = "1.8.29";
+  name = "htmldoc-${version}";
+  src = fetchurl {
+    url = "https://github.com/michaelrsweet/htmldoc/releases/download"
+      + "/release-${version}/htmldoc-${version}-source.tar.gz";
+    sha256 = "15x0xdf487j4i4gfap5yr83airxnbp2v4lxaz79a4s3iirrq39p0";
+  };
+  buildInputs =
+    stdenv.lib.ifEnable stdenv.isDarwin [SystemConfiguration Foundation];
```
Comment (LnL7) [inline pkgs/tools/typesetting/htmldoc/default.nix]: use `stdenv.lib.optionals`, I think ifEnable is deprecated

**Review by LnL7 — APPROVED** [review]

**Pull request merged**